crates/optimism/trie/src/live.rs

```diff
@@ -110,6 +136,26 @@ where
         let write_trie_updates_duration = start.elapsed() - calculate_state_root_duration;
         let execute_and_store_total_duration = start.elapsed();
 
+        #[cfg(feature = "metrics")]
+        {
+            self.metrics.execution_duration_seconds.record(execute_block_duration);
+            self.metrics.state_root_duration_seconds.record(calculate_state_root_duration);
+            self.metrics.write_duration_seconds.record(write_trie_updates_duration);
+            self.metrics.total_duration_seconds.record(execute_and_store_total_duration);
+            self.metrics
+                .account_trie_updates_written_total
+                .increment(update_result.account_trie_updates_written);
+            self.metrics
+                .storage_trie_updates_written_total
+                .increment(update_result.storage_trie_updates_written);
+            self.metrics
+                .hashed_accounts_written_total
+                .increment(update_result.hashed_accounts_written);
+            self.metrics
+                .hashed_storages_written_total
+                .increment(update_result.hashed_storages_written);
+        }
```

Comment:
let's make intermediary types in metrics module
```rust
pub struct OperationDurations {
    pub execute_block: Duration,
    pub calculate_state_root: Duration,
    ...
}
```
and
```rust
pub struct WriteCounts {
    pub account_trie_updates: usize,
    ...
}
```
and then add two methods `OpProofsStorageWithMetrics::record_operation_durations(&self, durations: OperationDurations)` and `OpProofsStorageWithMetrics::increment_write_counts(&self, counts: WriteCounts)`